firewalldb/sql_migration.go

```diff
@@ -0,0 +1,478 @@
+package firewalldb
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/lightninglabs/lightning-terminal/db/sqlc"
+	"github.com/lightningnetwork/lnd/sqldb"
+	"go.etcd.io/bbolt"
+)
+
+// kvParams is a type alias for the InsertKVStoreRecordParams, to shorten the
+// line length in the migration code.
+type kvParams = sqlc.InsertKVStoreRecordParams
+
+// MigrateFirewallDBToSQL runs the migration of the firwalldb stores from the
+// bbolt database to a SQL database. The migration is done in a single
+// transaction to ensure that all rows in the stores are migrated or none at
+// all.
+//
+// Note that this migration currently only migrates the kvstores, but will be
+// extended in the future to also migrate the privacy mapper and action stores.
+//
+// NOTE: As sessions may contain linked sessions and accounts, the sessions and
+// accounts sql migration MUST be run prior to this migration.
+func MigrateFirewallDBToSQL(ctx context.Context, kvStore *BoltDB,
```

Comment:
same as for sessions PR, rather pass raw kvdb pointer in here & not the BoltDB

Reply:
note that you are already using the raw pointer only & not actually using the BboltDb at all